src/Application/Config.h

```diff
@@ -16,32 +16,31 @@
 //   limitations under the License.
 
 #include <string>
+#include "yaml-cpp/yaml.h"
+#include "yaml-cpp/node/node.h"
 
-
-class Configuration {
+class Config {
 private:
-    std::string         m_strBrokerURI;
-    int                 m_intServerSleepCycle = 15;
+    // Private constructor per singleton pattern
+    Config();
 
-protected:
-    // Constructor(s)
-    Configuration() {};
+    static Config *inst_;   // The one, single instance
 
-    // Destructor
-    ~Configuration() {};
+    // holds the contents of the config.yaml file
+    YAML::Node yamlConfig;
 
 public:
-    std::string&        BrokerURI = m_strBrokerURI;
-    int&                ServerSleepCycle = m_intServerSleepCycle;
+    // Config variables
```

Comment:
The members here shouldn't all be public, and should also be better formatted. The std::string& BrokerURI = m_strBrokerURI allows for convenience but with redirection and still protecting the members. Later if required we could hide more things behind the = or change BrokerURI to BrokerURI() if required.

Reply:
Done!